test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -38,9 +41,104 @@ async def test_default_tombstone_gc(manager: ManagerClient, rf: int, tablets: bo
         async with new_test_table(cql, keyspace, "p int primary key, x int", " with tombstone_gc = {'mode': 'disabled'}") as table:
             await cql.run_async(f"ALTER TABLE {table} add y int")
             check_tombstone_gc_mode(cql, table, "disabled")
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="issue #15607")
+async def test_group0_tombstone_gc(manager: ManagerClient):
+    """
+    Regression test for #15607.
+
+    Test that the tombstones are being cleaned after the group 0 catching up.
+
+    Test #1:
+
+    Arrange:
+        - start 3 nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are cleaned up eventually
+
+    Test #2:
+
+    Arrange:
+        - stop one of the nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are not cleaned up (one of the nodes is down, so they can't catch up)
+
+    Test #3:
+
+    Arrange:
+        - start the node again
+    Act:
+        - tombstones exist from the previous test
+    Assert:
+        - the tombstones are cleaned up eventually
+    """
+    cfg = {
+        'group0_tombstone_gc_interval_in_ms': 1000,  # this is 1 hour by default
+    }
+    servers = [await manager.server_add(config=cfg) for _ in range(3)]
+    logs = [await manager.server_open_log(srv.server_id) for srv in servers]
+    marks = [await log.mark() for log in logs]
+
+    cql = manager.get_cql()
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    # create/alter/drop a few tables
+    async def alter_system_schema():
+        for _ in range(5):
+            async with new_test_table(cql, keyspace, "p int primary key, x int") as table:
+                await cql.run_async(f"ALTER TABLE {table} add y int")
+
+    def has_tombstone(row):
+        if row.metadata is None:
+            return False
+        metadata = json.loads(row.metadata)
+        tombstone = metadata.get("tombstone")
+        return tombstone is not None
+
+    async def tombstone_gc_completed():
+        count_rows_tombstone_total = 0
+        for tbl in ("raft", "scylla_local"):
+            res = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS(system.{tbl})", host=hosts[0]))
+            count_rows_tombstone = sum(1 for row in res if has_tombstone(row))
+            logger.info(f"{tbl} tombstones: {count_rows_tombstone}, total: {len(res)}")
+            count_rows_tombstone_total += count_rows_tombstone
+        if count_rows_tombstone_total != 0:
+            return None
+        return True
+
+    async def verify_tombstone_gc(timeout=15):
+        return [await log.wait_for("group0_state_broadcaster - Setting repair time of group0 tables", mark, timeout=timeout) for log, mark in zip(logs, marks)]
```

Comment:
This should probably be a `debug` level message; you can enable debug-level logging for this logger when starting the servers.

Also I'd consider the message an implementation detail, so I'd maybe add this particular check (which is a white-box type of check) in a later commit (when you actually implement the feature)